sdk/couchbase/src/subdoc/macros.rs

```diff
@@ -0,0 +1,52 @@
+use serde::{Serialize, Serializer};
+
+pub enum LookupInMacros {
+    Document,
+    ExpiryTime,
+    CAS,
+    SeqNo,
+    LastModified,
+    IsDeleted,
+    ValueSizeBytes,
+    RevId,
+}
+
+impl From<LookupInMacros> for String {
+    fn from(macro_: LookupInMacros) -> Self {
+        match macro_ {
+            LookupInMacros::Document => "$document".to_string(),
+            LookupInMacros::ExpiryTime => "$document.exptime".to_string(),
+            LookupInMacros::CAS => "$document.CAS".to_string(),
+            LookupInMacros::SeqNo => "$document.seqno".to_string(),
+            LookupInMacros::LastModified => "$document.last_modified".to_string(),
+            LookupInMacros::IsDeleted => "$document.deleted".to_string(),
+            LookupInMacros::ValueSizeBytes => "$document.value_bytes".to_string(),
+            LookupInMacros::RevId => "$document.revid".to_string(),
+        }
+    }
+}
+
+pub enum MutateInMacros {
+    CAS,
+    SeqNo,
+    ValueCrc32c,
+}
+
+pub(crate) static MUTATE_IN_MACROS: &[&[u8]] = &[
+    br#""${Mutation.CAS}""#,
+    br#""${Mutation.seqno}""#,
+    br#""${Mutation.value_crc32c}""#,
+];
+
+impl Serialize for MutateInMacros {
```

Comment:
This can probably be done using the serde derive macros. There's a field level rename derive.

Reply:
Good point! Done